open_seq2seq/parts/rnns/utils.py

```diff
@@ -11,134 +11,163 @@
-                                          forget_bias=1.0),
-                                input_keep_prob=dp_input_keep_prob,
-                                output_keep_prob=dp_output_keep_prob)
-      else:  # residual connection required
-        if dp_input_keep_prob == 1.0 and dp_output_keep_prob == 1.0:
-          return ResidualWrapper(GLSTMCell(num_units=size,
-                                           number_of_groups=num_groups,
-                                           num_proj=proj_size,
-                                           forget_bias=1.0))
-        else:
-          return ResidualWrapper(DropoutWrapper(
-            GLSTMCell(
-              num_units=size,
-              number_of_groups=num_groups,
-              num_proj=proj_size,
-              forget_bias=1.0,
-            ),
-            input_keep_prob=dp_input_keep_prob,
-            output_keep_prob=dp_output_keep_prob,
-          ))
-    elif cell_type == "slstm":
-      if not residual_connections:
-        if dp_input_keep_prob == 1.0 and dp_output_keep_prob == 1.0:
-          return BasicSLSTMCell(num_units=size)
-        else:
-          return DropoutWrapper(BasicSLSTMCell(num_units=size),
-                                input_keep_prob=dp_input_keep_prob,
-                                output_keep_prob=dp_output_keep_prob
-                                )
-      else:  # residual connection required
-        if dp_input_keep_prob == 1.0 and dp_output_keep_prob == 1.0:
-          return ResidualWrapper(BasicSLSTMCell(num_units=size))
-        else:
-          return ResidualWrapper(DropoutWrapper(
-            BasicSLSTMCell(num_units=size),
-            input_keep_prob=dp_input_keep_prob,
-            output_keep_prob=dp_output_keep_prob,
-          ))
-    else:
-      raise ValueError("Unknown RNN cell class: {}".format(cell_type))
+     Args:
+       cell_class: Tensorflow RNN cell class
+       cell_params (dict): cell parameters
+       dp_input_keep_prob (float): (default: 1.0) input dropout keep probability
+       dp_output_keep_prob (float): (default: 1.0) output dropout keep probability
+       residual_connections (bool): whether to add residual connection
 
-  if num_layers > 1:
-    if wrap_to_multi_rnn:
-      return MultiRNNCell([single_cell(cell_params) for _ in range(num_layers)])
-    else:
-      cells = []  # for GNMT-like attention in decoder
-      for i in range(num_layers):
-        cells.append(single_cell(cell_params))
-      return cells
-  else:
-    return single_cell(cell_params)
+     Returns:
+       TF RNN instance
+    """
+  cell = cell_class(**cell_params)
+  if residual_connections:
+    cell = ResidualWrapper(cell)
+  if dp_input_keep_prob != 1.0 or dp_output_keep_prob != 1.0:
+    cell = DropoutWrapper(cell, input_keep_prob=dp_input_keep_prob,
+                          output_keep_prob=dp_output_keep_prob)
+  return cell
+
+
+
+# def create_rnn_cell(cell_type,
```

Comment:
Also maybe remove unnecessary comments (if that code will be needed later it can be retrieved from the commit history)?